src/Form/Layout.php

```diff
@@ -169,7 +169,13 @@ protected function recursiveRender(): void
             }
 
             $template = $element->renderLabel ? $labeledControl : $noLabelControl;
-            $label = $element->caption ?? $element->entityField->getField()->getCaption();
+            $label = $element->caption;
+            if ($label === null) {
+                $label = $element->entityField->getField()->getCaption();
+                if (property_exists($element, 'model')) {
+                    $label = preg_replace('~ ID$~i', '', $label);
```

Comment:
TODO replace also "their reference field name" like in https://github.com/atk4/data/pull/1239/commits/d639bc958e5bf2d2f6827a1acbf7527101d565ee#diff-1c01b02e4e8021fe40414d7afd3f719df174d58c7076e2d436e8d994eef8f603R71